plan/physical_plan_builder.go

```diff
@@ -1176,3 +1120,43 @@ func (p *Distinct) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanIn
 	p.storePlanInfo(prop, info)
 	return info, nil
 }
+
+// PhysicalInitializer will do the initialization of some attributes after the convert2Physical process finish.
+type physicalInitializer struct {
+	ctx       context.Context
```

Comment:
If change the input to `Plan` interface, we can use this to set correlated for logical plans.